DataGenerator/DataGenerator.py

```diff
@@ -18,180 +18,222 @@
+        self.config = kwargs['config']
+        self.keys = kwargs['keys']
+        # self.label            = label
+        self.inference = kwargs['inference']
+        self.PatientList = PatientList
+        # self.n_norm = kwargs['numerical_norm']
+        # self.c_norm = kwargs['category_norm']
+        self.config = kwargs['config']
 
     def __len__(self):
-        return int(self.mastersheet.shape[0])
+        return int(len(self.PatientList))
 
     def __getitem__(self, id):
 
         datadict = {}
         roiSize = [10, 40, 40]
-        AnatomyPath = PatientList[i].scan.experiment['CT'].path()
-        
-        #label    = self.mastersheet[self.label].iloc[id]
+        patient_id = self.PatientList[id].label
+        ScanPath = self.config['DATA']['DataFolder'] + patient_id + '\\' + patient_id + '\\' + 'scans\\'
+        subfolder_list = os.listdir(ScanPath)
+        reader = image_reader.ITKReader()
+        label = self.PatientList[id].fields['survival_months']
         # Get the mask of PTV
-        if "Dose" in self.keys or "Anatomy" in self.keys:
-            if self.config['DATA']['Use_mask']:
-                path = self.mastersheet["CTPath"].iloc[id]
-                mask = path[0:-16] + 'structs\PTV.nrrd'
-                mask_img = LoadImg(mask)
-                properties = regionprops(mask_img.astype(np.int8), mask_img)
-                cropbox = properties[0].bbox
+        # if "Dose" in self.keys or "Anatomy" in self.keys:
+        #     if self.config['DATA']['Use_mask']:
+        #         RT_match_folder = [match for match in subfolder_list if "Structs" in match]
+        #         full_RT_path = ScanPath + RT_match_folder[0] + '\\resources\\secondary\\files\\1-1.dcm'
+        #         # read the rtstruct
+        #
+        #         # need process to convert 2D points to 3D masks
+        #         properties = regionprops(mask_img.astype(np.int8), mask_img)
+        #         cropbox = properties[0].bbox
 
         if "Dose" in self.keys:
-            dose     = LoadImg(self.mastersheet["DosePath"].iloc[id])
-            #maxDoseCoords = findMaxDoseCoord(dose) # Find coordinates of max dose
-            #checkCrop(maxDoseCoords, roiSize, dose.shape, self.mastersheet["DosePath"].iloc[id]) # Check if the crop works (min-max image shape costraint)
-            #datadict["Dose"]  = np.expand_dims(CropImg(dose, maxDoseCoords, roiSize),0)
+            Dose_match_folders = [match for match in subfolder_list if "Dose" in match]
+            full_Dose_path = [ScanPath + Dose_match_folder + '\\resources\\DICOM\\files\\' for Dose_match_folder in
+                              Dose_match_folders]
+            if len(full_Dose_path) > 1:
+                for dose_path in full_Dose_path:
+                    itkObj = reader.read(dose_path + '1-1.dcm')
+                    dose, info = reader.get_data(itkObj)
+                    for value in info.values():
+                        if 'totalhetero' in value:
```

Comment:
This is one of my point, we shouldn't use hard coded value like 'totalhetero'. Perhaps a better way to describe the toal dose.